### PR TITLE
Update balance additional events

### PR DESCRIPTION
**Pull request by straticJeff**
Handles treasury.Awarded and balance.balanceSet across the 3 supported chains

### CODE REVIEW

File: prawns/balances/db/migrations/1649868972554-Data.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module.exports = class Data1649868972554 {
```
Comment (bdmason) [inline prawns/balances/db/migrations/1649868972554-Data.js:1]: there's only supposed the be 1 migrations file (they're not really migrations). Please delete them all and run yarn migration again

File: prawns/balances/schema.graphql [hunk 2]
```diff
@@ -57,3 +57,29 @@ type SubstrateTreasuryDeposit @entity {
   blockNumber: BigInt!
   date: DateTime!
 }
+
+type SubstrateTreasuryAwarded @entity {
+  "network:block:index"
+  id: ID!
+  depositor: SubstrateBalanceAccount!
```
Comment (bdmason) [inline prawns/balances/schema.graphql:64]: maybe change this name to depositee (if that's even a word!) or receiver

File: prawns/balances/src/handlers/balances.balanceset.event.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  SubstrateNetwork,
+  SubstrateTreasuryAwarded,
+} from '../model';
+import { getBalancesBalanceSetEvent } from './typeGetters/getBalancesEvents';
+
+export default (network: SubstrateNetwork, tokenIndex: number) => {
+  return async (ctx: EventHandlerContext) => {
+
+    const depositor = ctx.extrinsic?.signer;
+    if (!depositor) return;
```
Comment (jdomingos) [inline prawns/balances/src/handlers/balances.balanceset.event.ts:14]: If I'm not wrong, the signer is not the `depositor`, it's just the caller - which is the root.

Do we really need this check here? We don't even use the extrinsic or the signer.

I suggest to remove it or at most do:
```
if (ctx.extrinsic?.signer) return;
```

If you **really** wanna keep it.

File: prawns/balances/src/handlers/typeGetters/getTreasuryEvents.ts [hunk 4]
```diff
@@ -46,3 +55,66 @@ export function getTreasuryDepositEvent(
...
+        };
+      } else if (event.isV9160) {
+        return event.asV9160;
+      } else {
+        return event.asLatest;
+      }
+    }
+
+    default: {
+      throw new Error('getBalancesWithdrawEvent::network not supported');
```
Comment (jdomingos) [inline prawns/balances/src/handlers/typeGetters/getTreasuryEvents.ts:117]: It should be:
`getTreasuryAwardedEvent::network not supported`

😄

**Review by jdomingos — COMMENTED** [review]
Looks good to me overall, just small things that in my opinion should be changed.

Also agree with the migration comment from @bdmason!

Nice job 👏

File: prawns/balances/db/migrations/1649868972554-Data.js (hunk 1 above)
Comment (markdavison) [reply prawns/balances/db/migrations/1649868972554-Data.js:1]: The devkit cli tool has a command to do it for you.

`devkit scaffolding balances`

Comment (markdavison) [reply prawns/balances/db/migrations/1649868972554-Data.js:1]: Probably needs a better name than scaffolding ... maybe schema:create

File: prawns/balances/src/handlers/balances.balanceset.event.ts (hunk 3 above)
Comment (bdmason) [reply prawns/balances/src/handlers/balances.balanceset.event.ts:14]: Good spot about the signer, it's kind of redundant here.

FYI, this check in our other handlers is more for TypeScript convenience, as the extrinsic isn't a required property on the event type, event though in all our use cases it is guaranteed to be there.

File: prawns/balances/src/handlers/typeGetters/getTreasuryEvents.ts (hunk 4 above)
Comment (bdmason) [reply prawns/balances/src/handlers/typeGetters/getTreasuryEvents.ts:117]: eagle eyes

File: prawns/balances/db/migrations/1649868972554-Data.js (hunk 1 above)
Comment (bdmason) [reply prawns/balances/db/migrations/1649868972554-Data.js:1]: There's something I did in the EVM work that handles this for you, I've removed it now I've moved away from postgres and don't need migrations though. We can gitignore all the models, types & migrations code which forces a fresh build//scaffold of everything. Makes the PRs a lot nicer too. When we take on either governance or container orchestration in the squid I think it should be preceded with a little tech debt cleanup in the prawns workflow, which will have a knock on effect in the cli package

Comment (straticJeff) [reply prawns/balances/db/migrations/1649868972554-Data.js:1]: Thanks guys. I'll regenerate this. 

What's the underlying reason why we only have one migration? I know we're not migrating as such (right now), does this just make things simpler and/or help avoid conflicts somehow?

File: prawns/balances/src/handlers/balances.balanceset.event.ts (hunk 3 above)
Comment (straticJeff) [reply prawns/balances/src/handlers/balances.balanceset.event.ts:14]: Yep, I understand now this is redundant (by virtue of it being done under a sudo call) so will remove it. Cheers guys.

Comment (jdomingos) [reply prawns/balances/src/handlers/balances.balanceset.event.ts:14]: @straticJeff, you ended up not removing it. Is it because you forgot to push? :D

**Review by bdmason — APPROVED** [review]

**Review by jdomingos — APPROVED** [review]

**Pull request merged**